[DNM] Lima II: Multi-Z Maintenance Shenanigans Turbo Championship Edition

sqnztb opened the pull request:
~~nice~~

Just merge it, what's the worst that can happen?

![image](https://user-images.githubusercontent.com/28007787/128641665-fcd35690-d50b-418c-82cb-2e683c27d3bc.png)

... heh.

Well everyone. Here she is. Lima II: Multi-Z Maintenance Shenanigans Turbo Championship Edition (L2MZMSTCE for short.)

Let's go over the big changes.




**Medical**
- Medical gets some patient recovery rooms. These are semi-private. Close the blinds and the patient cannot be seen by passersby in the hall.
- Morgue shifts down a bit to compensate.
- Viro gets the Z.
- Chemistry gets the Z.

Overview: 
![image](https://user-images.githubusercontent.com/28007787/128641831-76fa3b40-a064-45b2-9774-070db50b47da.png)
Lower Viro:
![image](https://user-images.githubusercontent.com/28007787/128641860-5108814b-70f6-4e93-addb-bd08e4a81b5d.png)
Lower Chem:
![image](https://user-images.githubusercontent.com/28007787/128641865-d2efb1b9-41d0-43ec-9809-558d98b6a735.png)




**Science**
- Xenobio gets the Z
- New lower-z testing lab.

Overview, note the stairs at the top of the image.
![image](https://user-images.githubusercontent.com/28007787/128641910-1ef22d41-211e-4b9a-9bb2-2b225d23ae66.png)
Lower Xenobio:
![image](https://user-images.githubusercontent.com/28007787/128641954-91b5f27e-ef46-42ab-9dd8-5e8dc2b95509.png)
Lower testing lab:
![image](https://user-images.githubusercontent.com/28007787/128641963-6c37d8a8-0c5c-4b77-9bda-4c99c506f0de.png)




**Service**
Nearly everything in service got touched in some ways. Y'all got a nice pretty service hall now though.
![image](https://user-images.githubusercontent.com/28007787/128642063-37c480d4-8de1-440c-a430-acecf2d91386.png)
Botany bee dungeon.
![image](https://user-images.githubusercontent.com/28007787/128642091-3fe8adf2-7549-4a0e-8e90-70b05be37488.png)





**Engineering/Atmospherics**
HFR goes to lower level. Some of the engineering equipment goes down to the lower level. New breakroom.
![image](https://user-images.githubusercontent.com/28007787/128642123-ecb5ee21-9e37-42bc-a83b-7c170740dbeb.png)



**Maintenance Changes**
Where applicable, I have changed some of the location-named maintenance area types to department-named maintenance types. E.g. "Starboard Maintenance" to "Science Maints", "Aft Maints" to "Cargo Maints", etc.



And finally, here's the map of the lower Z.
![image](https://user-images.githubusercontent.com/28007787/128642267-4e0a0a8c-4919-4991-8b53-57b5b68b26ca.png)


Feedback is requested prior to merging this. It's not 100% ready in its current state, but it's close enough to start the pull and get the community feedback. I still have a little bit of detailing to do.

Jolly-66 commented:
You know there are sub typed department maints right? Check Pubby for ref.

sqnztb commented:
Yes. I changed several maint areas to them.
> Maintenance Changes
Where applicable, I have changed some of the location-named maintenance area types to department-named maintenance types. E.g. "Starboard Maintenance" to "Science Maints", "Aft Maints" to "Cargo Maints", etc.

![image](https://user-images.githubusercontent.com/28007787/128642649-9c78e0a1-d10d-468d-83e6-f3b8457c3052.png)

Jolly-66 commented:
> Yes. I changed several maint areas to them.
> 
> > Maintenance Changes
> > Where applicable, I have changed some of the location-named maintenance area types to department-named maintenance types. E.g. "Starboard Maintenance" to "Science Maints", "Aft Maints" to "Cargo Maints", etc.
> 
> ![image](https://user-images.githubusercontent.com/28007787/128642649-9c78e0a1-d10d-468d-83e6-f3b8457c3052.png)

Yeah I was half asleep reading the pr my bad

MrMelbert commented:
![image](https://user-images.githubusercontent.com/51863163/128648362-3865452f-e02b-4028-a021-e3d747e03d19.png)
![image](https://user-images.githubusercontent.com/51863163/128648371-d506a0fc-3d21-4fb9-bdfc-eb9b117b186d.png)

MrMelbert commented:
![image](https://user-images.githubusercontent.com/51863163/128648491-82a2339f-8f0c-46bc-93d1-4b19d29c443e.png)

sqnztb commented:
Yeah I was working down the list for the test station z issues, was getting into the RC section. Everything else has been fixed.

sqnztb commented:
Lots of little cleanup. Decals that were hidden underneath wall turfs. Air alarms, newscasters, status displays. Science breakroom. Double checked all of the piping and wiring. Whiteship dock is now properly big enough to fit the frigate (delta) and salvage ship (meta) whiteships, though I don't give two shits about the 13x41 monstrosity of tram's whiteship, fuck that noise.

Some slight non-modularity with the nav computer tweak, though I don't expect this to run into any merge conflicts from tg any time soon. Still researching this so I'm leaving it up as DNM. 

It's ready though. I hope.

Jolly-66 commented:
Did you doc the modular changes?

Jolly-66 commented on code/modules/shuttle/navigation_computer.dm:
```diff
@@ -20,7 +20,7 @@
 	var/x_offset = 0
 	var/y_offset = 0
-	var/list/whitelist_turfs = list(/turf/open/space, /turf/open/floor/plating, /turf/open/lava)
+	var/list/whitelist_turfs = list(/turf/open/space, /turf/open/floor/plating, /turf/open/lava, /turf/open/openspace)
```
I'm going to kill you

Add this to the Jollystation readme

sqnztb replied:
>  Still researching this so I'm leaving it up as DNM.

Also this got discussed separately in coderbus, I've submitted a pull to tg to fix this issue. I'll revert that file on this when it merges.

MrMelbert commented:
Science

This room's very plain, not sure if anything else can be done here but eh
![image](https://user-images.githubusercontent.com/51863163/128776686-07832e46-398f-443b-824d-fbc5417f009d.png)


Plating
![image](https://user-images.githubusercontent.com/51863163/128776634-aaa3c18a-c281-445b-8607-e86ffda58c58.png)
Fire door
![image](https://user-images.githubusercontent.com/51863163/128776658-9f5e7517-b4db-4280-b83e-d68eb7c8bef5.png)
One sliver of intercom dead zone, not too important
![image](https://user-images.githubusercontent.com/51863163/128776757-23387ee6-bab0-48ea-9d62-1dc2fb8765d4.png)
Fire door
![image](https://user-images.githubusercontent.com/51863163/128776773-f5446c45-f71b-4541-b2ee-d7a749fe22e8.png)

MrMelbert commented:
Engineering

Seems like none of these doors have firedoors at all
![image](https://user-images.githubusercontent.com/51863163/128776925-1921ab70-d554-4590-9e11-e64413001021.png)

MrMelbert commented:
Service

Same lack of firedoors around the service hall
![image](https://user-images.githubusercontent.com/51863163/128777154-4cc670a1-9208-434f-a0be-02de440ec926.png)
This far end has no intercom coverage, unsure if intentional
![image](https://user-images.githubusercontent.com/51863163/128777184-f38d34bd-55c6-4dad-99d0-8528ba74e214.png)

MrMelbert commented:
Medbay 

This disposals outlet leads riiight into a grille
![image](https://user-images.githubusercontent.com/51863163/128777469-4f4e7521-8889-45f8-87f2-94820a1fee27.png)

MrMelbert commented:
Other misc notes:

- No carp spawn on the lower levels?
- (Major) Space dust is totally going to make the lower maintenance uninhabitable very often
- The entire lower area being the same area might very issues with area based things, it might be wise to break it up into sections, maybe define some new maintenance areas in the code if you have to.
- Some of the minor fixes and adjustments made to Lima prior didn't make it over to this version, but i'll just port them over myself when this is done and dusted

overall, very gud

MrMelbert reviewed (changes requested):
the jannie's back door is a space area
![image](https://user-images.githubusercontent.com/51863163/128779354-27195495-4033-4876-a56d-451882a7bdf7.png)

Jolly-66 reviewed (changes requested):
Add a trap door to the HoP line.

Trust me this'll be funny.

sqnztb commented:
I'm not going to let you errantly space people from the line.

Jolly-66 commented:
> I'm not going to let you errantly space people from the line.

Just dump them in maint like the rats they are damnit!!!!!

Jolly-66 commented:
Oh yeah don't forget to add the closes thing to close that one rogue lima bug report

sqnztb commented:
Cleaned up all of the windoors across the station. No more var/iconstate edited garbage like...
```/obj/machinery/door/window/northleft{
    base_state = "right";
    dir = 2;
    icon_state = "right";
    name = "Containment Pen #4";
    req_access_txt = "55"
    },```

Also fixed all of the lower z's grille armor. I'd copied the wrong turf (one that was just grille+openspace instead of grille+rwall) so now space dust shouldn't be anywhere near as bad. Nor should you be able to fall down to central command.

And thank you for reminding me.
Fixes #63

Jolly-66 commented:
Pending upstream merge; some fixes and updates will come after that

@MrMelbert
